.eslintrc.js

```diff
@@ -2,20 +2,55 @@ module.exports = {
     root: true,
     parser: '@typescript-eslint/parser',
     plugins: [
-        '@typescript-eslint',
-        'mocha'
+        '@typescript-eslint'
     ],
     extends: [
-        'eslint:recommended',
         'plugin:@typescript-eslint/recommended',
     ],
+    // eslint-config-preact needs a Jest version to be happy, even if Jest isn't used.
+    // See https://github.com/preactjs/eslint-config-preact/issues/19#issuecomment-997924892
+    settings: {
+        jest: { "version": 27 },
+    },
     rules: {
         "@typescript-eslint/explicit-module-boundary-types": "off",
         "camelcase": ["error", { "properties": "never", "ignoreDestructuring": true }],
         "no-console": "error"
     },
-    "env": {
-        "node": true,
-        "es6": true,
+    env: {
+        node: true,
+        es6: true,
     },
+    overrides: [
+        {
+            files: ["test/**/*.ts"],
+            parser: '@typescript-eslint/parser',
+            plugins: [
+                '@typescript-eslint',
+                'mocha',
+            ],
+        },
+        {
+            files: ["web/**/*.ts", "web/**/*.tsx"],
+            parser: '@typescript-eslint/parser',
+            env: {
+                browser: true,
+                node: false,
+            },
+            extends: [
+                'plugin:@typescript-eslint/recommended',
+                'preact',
+            ],
+            plugins: [
+                '@typescript-eslint',
+            ],
+            rules: {
+                "no-console": "off",
+                "no-unused-vars": "off",
+                "@typescript-eslint/no-unused-vars": ["error"],
+                "no-useless-constructor": "off",
+                "@typescript-eslint/no-useless-constructor": ["error"],
```

Comment:
I couldn't figure out why the regular eslint rules were still being used.
Only the TypeScript variants make sense for this code base.

Reply:
Hm, suppose we can come back to this one when it poses a problem for us. The linter seems happy for the moment.